### PR TITLE
fix: surface branch_messages constraint violations instead of INSERT OR IGNORE

### LINKED ISSUE #158: bug: INSERT OR IGNORE on branch_messages hides all constraint violations, not just duplicates
## Summary

Both writers of `branch_messages` use `INSERT OR IGNORE`, which was intended to absorb the duplicate-primary-key case but in fact suppresses every constraint class — NOT NULL, FOREIGN KEY, CHECK. This is the mechanism that made #155 invisible for roughly a week: every insert failed, nothing raised, nothing logged, and the suite stayed green.

## Evidence

Two sites:

- `src/ccrecall/branch_ops.py:187` (`diff_branch_messages`)
- `src/ccrecall/hooks/backfill_tool_content.py:539`

Both:

```sql
INSERT OR IGNORE INTO branch_messages (branch_id, message_id) VALUES (?, ?)
```

Confirmed behavior — a NOT NULL violation is swallowed with exit code 0 and no diagnostic:

```
$ sqlite3 :memory: "CREATE TABLE t (a INTEGER NOT NULL, b INTEGER NOT NULL, pos INTEGER NOT NULL, PRIMARY KEY(a,b));
                    INSERT OR IGNORE INTO t (a,b) VALUES (1,2);
                    SELECT COUNT(*) FROM t;"
0
```

In #155 this produced a week-long total search outage with no error, no log line, and no failing test. The fix for the missing column addresses that one instance; the mechanism that hid it remains at both call sites and will hide the next constraint drift identically.

## Suggested fix

Pick one, applied to both sites:

1. **Narrow the suppression.** `INSERT OR IGNORE` is only wanted for "this link already exists." Since `diff_branch_messages` already computes `to_add = desired - existing`, the duplicate case is largely designed out and a plain `INSERT` would surface real violations. Where a race can still produce a duplicate, catch `sqlite3.IntegrityError` and re-raise anything that is not a uniqueness violation.
2. **Verify the write.** Keep `OR IGNORE`, but compare `cursor.rowcount` (or a follow-up `COUNT(*)`) against the intended row count and log an ERROR when they disagree. Cheaper to implement, and it converts a silent failure into a signal.

Option 1 is stronger — it fails at the point of damage instead of reporting after. Option 2 is a reasonable interim if there is a real duplicate-insert path that has to stay tolerated.

Either way this should also be reflected in a test that inserts against a `branch_messages` carrying an unexpected NOT NULL column and asserts the write is not silently dropped. No current test can catch this, because every test database is built from `SCHEMA_CORE` and therefore never diverges from what the code expects.

## Related

- #155 — the instance this mechanism concealed
- Companion issue on the unguarded `_migrate_to_v2` delete, which consumes `branch_messages` as a trusted input

### PR DESCRIPTION
**Pull request by NodeJSmith**
## Summary

Both writers of `branch_messages` used `INSERT OR IGNORE`, which suppresses **every** constraint class — NOT NULL, FOREIGN KEY, CHECK — not just the duplicate-link case it was meant to absorb. This is the mechanism that made #155 invisible for about a week: every link insert failed silently, nothing raised, nothing logged, and the suite stayed green.

## Fix

Option 1 from the issue (fail at the point of damage):

- New `insert_branch_message_links()` in `branch_ops.py`: a plain `INSERT` per link that absorbs only `UNIQUE constraint failed` (a concurrent-writer race, logged at WARNING) and re-raises every other `IntegrityError`.
- Both call sites now use it: `diff_branch_messages()` (which already pre-computes the missing set, so duplicates are designed out) and the tool-content backfill's link pass in `hooks/backfill_tool_content.py`.

## Tests

New `tests/test_branch_ops.py` (RED commit first, per repo convention):

- `test_diff_branch_messages_raises_on_constraint_violation` — rebuilds `branch_messages` with an unexpected NOT NULL column (the #155 drift shape) and asserts the write raises instead of vanishing. Failed before the fix (`DID NOT RAISE`).
- `test_insert_branch_message_links_absorbs_duplicate_links` — a pre-existing link is absorbed without raising, row count stays 1.
- `test_insert_branch_message_links_raises_on_foreign_key_violation` — an FK violation propagates.

Full suite: 1317 passed, 2 skipped. `uvx prek run --all-files`: all hooks pass.

Fixes #158

### CODE REVIEW

File: src/ccrecall/branch_ops.py [hunk 1]
```diff
@@ -185,9 +185,43 @@ def diff_branch_messages(
...
+    message_id: int,
+) -> None:
+    """Insert one branch_messages link; absorb a duplicate, raise anything else."""
+    try:
+        cursor.execute(
+            "INSERT INTO branch_messages (branch_id, message_id) VALUES (?, ?)",
+            (branch_db_id, message_id),
+        )
+    except sqlite3.IntegrityError as exc:
+        if "UNIQUE constraint failed" not in str(exc):
```
[the remark this answers is not in the document]
Comment (NodeJSmith) [reply src/ccrecall/branch_ops.py:220]: Checked against the actual schema: branch_messages only has PRIMARY KEY (branch_id, message_id) — no other unique constraint, and no trigger on the table either. The current suppression can only ever match that one constraint, so this isn't reachable today. Leaving as-is; will revisit if the schema ever grows a second unique rule on this table.
